stable/restore/files/nuorestore

```diff
@@ -86,49 +101,105 @@ do
            auto="$1"; shift;;
         "--auto="* )
            auto="${opt#*=}";;
+        "--labels" )
+           labels="$1"; shift;;
+        "--labels="* )
+           labels="${opt#*=}";;
+        "--archive-ids" )
+           archive_ids="$1"; shift;;
+        "--archive-ids="* )
+           archive_ids="${opt#*=}";;
+        "--manual" )
+           manual="$1"; shift;;
+        "--manual="* )
+           manual="${opt#*=}";;
         *) echo >&2 "Invalid nuorestore option: $opt ($@)"; exit 1;;
    esac
 done
 
 restore_type=$(echo $restore_type | tr '[:upper:]' '[:lower:]')
 auto=$(echo $auto | tr '[:upper:]' '[:lower:]')
 
-# set the persistent flag for the backup
-log "restore_type=$restore_type; writing $restore_source to $NUODB_RESTORE_REQUEST_PREFIX/$db_name/restore"
 
-nuocmd --api-server $NUOCMD_API_SERVER set value --key $NUODB_RESTORE_REQUEST_PREFIX/$db_name/restore \
-    --value $restore_source \
-    --unconditional
+if isRestoreRequestSupported; then
+   # Use nuodocker to place restore request
+   additional_args=""
+   restore_source_encoded="$(printf "%s" "${restore_source}" | base64 -w 0)"
+   user_data="restore_source_encoded='${restore_source_encoded}'"
+   [ -n "$labels" ] && additional_args="${additional_args} --labels ${labels}"
+   [ -n "$archive_ids" ] && additional_args="${additional_args} --archive-ids ${archive_ids}"
+   [ "$manual" == "true" ] && additional_args="${additional_args} --manual"
+   if [ -n "$restore_credentials" ]; then
+      restore_credentials_encoded="$(printf "%s" "${restore_credentials}" | base64 -w 0)"
+      user_data="${user_data};restore_credentials_encoded='${restore_credentials_encoded}'"
+   fi
 
-if [ -n "$restore_credentials" ]; then
-    nuocmd --api-server $NUOCMD_API_SERVER set value --key $NUODB_RESTORE_REQUEST_PREFIX/$db_name/restore/credentials \
-        --value $restore_credentials \
-        --unconditional
-fi
+   log "restore_type=${restore_type}; restore_source=${restore_source}; arguments=${additional_args}"
+   nuodocker request restore \
+      --db-name "$db_name" \
+      --type "$restore_type" \
+      --user-data "${user_data}" $additional_args
+   rc=$?
+   if [ $rc -ne 0 ]; then
+      echo "Restore request failed" >&2
+      exit $rc
+   fi
 
-log "Setting flag for database $db_name to restore from $restore_source on next database/SM startup"
+   if [ "$auto" = "true" ]; then
+      if [ "$restore_type" = "database" ]; then
+         log "restore.autoRestart=true - initiating full database restart for database $db_name"
+         nuocmd shutdown database --db-name "$db_name"
+      else
+         for archive_id in $(nuodocker get restore-requests --db-name "$db_name" | grep '[0-9]\+ automatic' | awk '{print $1}'); do
+            start_id=$(nuocmd show domain \
+               --process-format "###{archive_id::-1} {start_id}" | grep "###${archive_id}" | awk '{print $2}')
```

Comment:
Can you grep on `###$start_id ` (note the space)? Otherwise we can match an archive ID that is a prefix of `$start_id`, e.g. we want the running SM on `archive_id=1`, but this matches the one on `archive_id=10`.

Reply:
Good point! Thanks.